### PR TITLE
Implement interface `ClusterWaitState` for test framework;

### PR DESCRIPTION
**Pull request by cnutshell**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [ ] Improvement
- [ ] Documentation
- [x] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #4003

## What this PR does / why we need it:
- Implement interface `ClusterWaitState` which is used to wait for some expected state;
- Add more convenient methods within interface `ClusterState`;
- Add unit test, and make all unit tests runnable;

### CODE REVIEW

File: pkg/tests/service/errors.go [hunk 1]
```diff
@@ -19,6 +19,7 @@ import "fmt"
 var (
 	ErrServiceNotExist     = fmt.Errorf("service not exist")
 	ErrServiceNotStarted   = fmt.Errorf("service not started")
+	ErrStoreNotReported    = fmt.Errorf("store not reported")
 	ErrInvalidServiceIndex = fmt.Errorf("invalid service index")
 	ErrFailAllocatePort    = fmt.Errorf("fail to allocate port")
```
Comment (zhangxu19830126) [inline pkg/tests/service/errors.go:24]: why use fmt.Errorf ?  There has no any parameter

File: pkg/tests/service/options.go [hunk 2]
```diff
@@ -143,6 +145,17 @@ func (opt *Options) validate() {
...
 }
 
+// BuildHAKeeperConfig returns hakeeper.Config
+//
+// We could check timeout for dn/log store via hakeeper.Config.
+func (opt Options) BuildHAKeeperConfig() hakeeper.Config {
+	return hakeeper.Config{
+		TickPerSecond:   opt.hakeeper.tickPerSecond,
+		LogStoreTimeout: opt.hakeeper.logStoreTimeout,
+		DnStoreTimeout:  opt.hakeeper.dnStoreTimeout,
```
Comment (zhangxu19830126) [inline pkg/tests/service/options.go:155]: Use DNStoreTimeout

**Review by zhangxu19830126 — CHANGES_REQUESTED** [review]

File: pkg/tests/service/options.go (hunk 2 above)
Comment (cnutshell) [reply pkg/tests/service/options.go:155]: DnStoreTimeout is a field of `hakeeper.Config` which is defined within package `github.com/matrixorigin/matrixone/pkg/hakeeper`.

File: pkg/tests/service/errors.go (hunk 1 above)
Comment (cnutshell) [reply pkg/tests/service/errors.go:24]: Be used to using `fmt.Errorf`, just personal habit.
Do I need to replace `fmt.Errorf` with `errors.New`?

**Review by zhangxu19830126 — APPROVED** [review]

**Review by lni — APPROVED** [review]

**Pull request merged**